app/actions/remote/entry/common.ts

```diff
@@ -553,14 +555,29 @@ export async function handleEntryAfterLoadNavigation(
                     await setCurrentTeamAndChannelId(operator, initialTeamId, initialChannelId);
                 }
             }
-        } else if (currentChannelIdAfterLoad !== initialChannelId) {
+        } else if (currentChannelIdAfterLoad && currentChannelIdAfterLoad !== initialChannelId) {
```

Comment:
@enahum, When we navigate to GlobalThreads and then to a thread screen, after WebSocket reconnect, I am getting `Removed from the channel` Alert.

`currentChannelIdAfterLoad` is empty and `initialChannelId` is `GlobalThreads`.

That's why added the check. Would this affect the logic?

Reply:
@larkox looks like @anurag6713 found the issue I was unabke to replicate :)

Reply:
Just to be sure. When you are on a thread screen, the `currentChannelId` is empty? Is it not the one from the thread?

Reply:
Depends on how you go to the thread screen.

If you are going through the Global Threads screen and then to an individual thread, it's empty. 

If you are in a channel and then on a thread screen, then it's not empty.

Reply:
Then we have several problems here (some of them may be already solved, so bear with me).
- The most related to this PR is regarding being kicked out of a channel. If when the websocket reconnects we are no longer members of the channel where this thread was originated, we should be kicked out to the root with some message. I was expecting that to be solved because I assumed the "currentChannelId" was also set to the channel of the thread. Since that is not the case, this may break (or may be solved in a different way) because getting kicked out of a channel should trigger, among other things, the removal of the post from the database, which probably will lead to a weird state.
- This one would be a different ticket, but in terms of integrations (slash commands and apps in general) we may be using the currentChannelId for many things (filling up the slash command context and almost everything for apps (from fetching to executing them)). If we don't have a currentChannelId, we have to make sure those are using the right value (I guess the channel id in the root post object)

Does this make sense?